### PR TITLE
Java 15 and rpm/deb packaging

### LINKED ISSUE #1240: Upgrade to java 11+
I suggest we:
- [x] migrate to openjdk 14 and bundle it in the distribution
- [x] build and package anet distribution as rpm,msi,deb and container

### PR DESCRIPTION
**Pull request by VassilIordanov**
### Release notes

ANET now uses Java 15 and builds installer packages that contain a jre

Closes #1240 

#### System admin changes
- a Java JRE is no longer an external dependency
- ANET no longer supported as a service on a Windows server 

- [ ] anet.yml needs change
- [ ] db needs migration
- [x] documentation has changed
- [ ] graphql schema has changed

### Checklist
  - [x] Described the user behavior in PR body
  - [x] Referenced/updated all related issues
  - [x] commits follow a `repo#issue: Title` title format and [these 7 rules](https://chris.beams.io/posts/git-commit/)
  - [x] commits have a [clean history](https://epage.github.io/dev/commits/), otherwise PR may be squash-merged
  - [ ] Added and/or updated unit tests
  - [ ] Added and/or updated e2e tests
  - [ ] Added and/or updated data migrations
  - [x] Updated documentation
  - [x] Resolved all build errors and warnings
  - [ ] Opened debt issues for anything not resolved here

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -39,6 +43,7 @@ jobs:
     - name: Autobuild
       uses: github/codeql-action/autobuild@v1
 
+
```
Comment (gjvoosten) [inline .github/workflows/codeql-analysis.yml:46]: Extra blank line.

File: build.gradle [hunk 2]
```diff
@@ -105,6 +107,9 @@ dependencies {
 	implementation 'org.apache.poi:poi:4.1.2'
 	implementation 'org.apache.poi:poi-ooxml:4.1.2'
 
+	// For Java 11
```
Comment (gjvoosten) [inline build.gradle:110]: Suggestion: `For Java 11+`

File: build.gradle [hunk 3]
```diff
@@ -514,10 +541,9 @@ distributions {
 			into('docs/') {
 				from('anet.yml') { rename('anet.yml', 'anet.yml.template') }
 				from('anet-dictionary.yml') { rename('anet-dictionary.yml', 'anet-dictionary.yml.template') }
-				from(markdownToHtml)
-			}
-			into('lib/') {
-				from('lib/sqljdbc_auth.dll')
+				from('docs/INSTALL.md')
+				from('docs/TROUBLESHOOT.md')
+				from('prepare-psql.sql')
```
Comment (gjvoosten) [inline build.gradle:546]: Also pack `prepare-mssql.sql` even though it's currently empty.

File: build.gradle [hunk 4]
```diff
@@ -530,16 +556,39 @@ startScripts {
...
+			into 'anet/docs'
+		}
+		from('scripts/anet.service') { into '/etc/systemd/system' }
+		from('docs/INSTALL.md') { into 'anet/docs' }
+		from('docs/TROUBLESHOOT.md') { into 'anet/docs' }
+		from('anet-dictionary.yml') {
+			rename('anet-dictionary.yml', 'anet-dictionary.yml.template')
+			into 'anet/docs'
+			}
+		from('prepare-psql.sql') {
```
Comment (gjvoosten) [inline build.gradle:583]: Also add `prepare-mssql.sql`.

File: build.gradle [hunk 5]
```diff
@@ -530,16 +556,39 @@ startScripts {
...
 	version = '12.14.1'
 	yarnVersion = '1.22.10'
 	download = true
 }
+
+ospackage {
+        release = '2'
+        os = LINUX
+
+		into '/opt'
```
Comment (gjvoosten) [inline build.gradle:570]: Fix indentation.

File: build.gradle [hunk 6]
```diff
@@ -530,16 +556,39 @@ startScripts {
...
+			rename('anet-dictionary.yml', 'anet-dictionary.yml.template')
+			into 'anet/docs'
+			}
+		from('prepare-psql.sql') {
+			into 'anet/docs'
+		}
+    }
+
+task distDeb(dependsOn: "jpackageImage",type: Deb) {
+		arch = 'amd64'
```
Comment (gjvoosten) [inline build.gradle:589]: Fix indentation.

File: build.gradle [hunk 7]
```diff
@@ -530,16 +556,39 @@ startScripts {
...
+}
+
+task distRpm(dependsOn: "jpackageImage",type: Rpm) {
+		arch = 'x86_64'
```
Comment (gjvoosten) [inline build.gradle:593]: Fix indentation.

File: docs/INSTALL.md [hunk 8]
```diff
@@ -23,7 +22,7 @@ and Semantic Extractions for Search`.
 	- Users are required to have a modern web browser (Mozilla Firefox, Google Chrome, Microsoft Edge or other with good HTML5 support). IE11 is currently supported although performance may degrade and support will be discontinued beyond Q3 2019
 	- A service manager, such as [NSSM](https://nssm.cc/) , can be used to install ANET as a service on Windows
 - **Network Accessibility**
-	- Users will access the Application Server over HTTP/HTTPS (`80`/`443`)
+	- Users will access the Application Server over HTTP/HTTPS (`443`)
```
Comment (gjvoosten) [inline docs/INSTALL.md:25]: `over HTTPS` only (if only port `443`)

**Review by gjvoosten — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 2 above)
Comment (VassilIordanov) [reply build.gradle:110]: probably for Java 9+, changed to that

File: build.gradle (hunk 5 above)
Comment (VassilIordanov) [reply build.gradle:570]: ah, tabs vs spaces...

**Review by gjvoosten — APPROVED** [review]

**Pull request merged**